intro-to-functions/defining-functions.md

```diff
@@ -0,0 +1,455 @@
+# Defining Functions
+
+## Introduction
+
+A function is a reusable chunk of code. Every function is defined with a name and some logic.
+
+If, in a given project, we want to **define** our own reusable chunks of code, then we can define our own functions. Then, any code that has access to our functions can invoke them. To define a function, we must follow specific Python syntax.
+
+## Learning Goals
+
+- Apply Single Responsibility Principle to functions
+- Define a function (function signature, function body, return)
+- Demonstrate calling functions with and without arguments/parameters
+
+## Vocabulary and Synonyms
+
+| Vocab | Definition | Synonyms | How to Use in a Sentence
+| --- | --- | --- | ---
+| Function | Lines of code (1 or more) that are related, grouped together, and named. Once defined, these lines of code are reusable and can be called over and over again. | Method | "I defined a function," "I used a function," "I wrote a function"
```

Comment:
Hmmm...

I tend to like pairing "methods" definition with classes, since that's a concrete example to observe.

Since I like pairing the "methods" definiton with that, I almost want to say "Let's just get rid of "methods" here and not add the methods definition"

What do you all think of that?

(PS: my intention of adding "methods" was to help suss out the colloquial-ness of the phrase, but I definitely prefer having accuracy over not, now! I'm into getting rid of the word "method" here for sure)